src/baseRunner.ts

```diff
@@ -30,23 +31,36 @@ export abstract class BaseRunner {
         return process.platform === 'win32';
     }
 
-    public runPlaybook(): void {
-        var playbook = vscode.window.activeTextEditor ? vscode.window.activeTextEditor.document.fileName : null;
-        vscode.window.showInputBox({ value: playbook, prompt: 'Please input playbook name', placeHolder: 'playbook', password: false })
-            .then((input) => {
-                if (input != undefined && input != '') {
-                    playbook = input;
-                }
+    public runPlaybook(playbook: string): void {
 
-                this._outputChannel.append('\n' + Constants.LineSeperator + '\nRun playbook: ' + playbook);
-                this._outputChannel.show();
-
-                if (!utilties.validatePlaybook(playbook, this._outputChannel)) {
-                    return;
-                }
+        if (!playbook) {
+            playbook = vscode.window.activeTextEditor ? vscode.window.activeTextEditor.document.fileName : null;
+            vscode.window.showInputBox({ value: playbook, prompt: 'Please input playbook name', placeHolder: 'playbook', password: false })
```

Comment:
so if we run playbook using context menu, should we ask again to reconfirm file name?

Reply:
no. that's the good part of context menu